### PR TITLE
💄 style: add feature flag to show model name instead of id

### PR DESCRIPTION
**Pull request by khanhduyvt0101**
#### 💻 变更类型 | Change Type

- [ ] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [x] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] 📝 docs
- [ ] 🔨 chore

#### 🔀 变更说明 | Description of Change

This adds a feature flag `model_name` that can be turned on to show model name instead of model id in UI.

### CODE REVIEW

**Comment by arvinxx** [comment]
Actually I perfer to use code for better understanding what's the model id underhood instead of model name.

because model name can be customized but model id should be consistent

**Comment by khanhduyvt0101** [comment]
> Actually I perfer to use code for better understanding what's the model id underhood instead of model name.
> 
> because model name can be customized but model id should be consistent

I converted into a feature flag, instead. The default remains the same. If user prefers, they can turn on the feature flag to enable this customization. What do you think @arvinxx?

**Comment by arvinxx** [comment]
> I converted into a feature flag, instead. The default remains the same. If user prefers, they can turn on the feature flag to enable this customization. 

Yeah, a feature flag is a better one,I'm glad to merge it. But it need test to cover the implement at first

**Comment by khanhduyvt0101** [comment]
> > I converted into a feature flag, instead. The default remains the same. If user prefers, they can turn on the feature flag to enable this customization.
> 
> Yeah, a feature flag is a better one,I'm glad to merge it. But it need test to cover the implement at first

I have updated UT to meet coverage standards, please merge it @arvinxx

**Comment by khanhduyvt0101** [comment]
Hi @arvinxx , I just updated base on your feedback

**Comment by arvinxx** [comment]
@khanhduyvt0101 hi, plz rebase the code

**Comment by khanhduyvt0101** [comment]
> @khanhduyvt0101 hi, plz rebase the code

I rebased base on your request @arvinxx

File: src/components/ModelTag/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { ModelTag as DefaultModelTag, ModelIcon, ModelTagProps } from '@lobehub/icons';
```
Comment (arvinxx) [inline src/components/ModelTag/index.tsx:3]: add a `'use client'` here

File: src/components/ModelTag/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { featureFlagsSelectors, useServerConfigStore } from '@/store/serverConfig';
+import { useUserStore } from '@/store/user';
+import { modelProviderSelectors } from '@/store/user/selectors';
+
+const CustomModelTag = memo<ModelTagProps>(({ type = 'mono', model, ...rest }) => {
+  const enabledList = useUserStore(modelProviderSelectors.modelProviderListForModelSelect);
+
+  let modelCard;
+
+  for (const provider of enabledList) {
```
Comment (arvinxx) [inline src/components/ModelTag/index.tsx:23]: If you want to display the model name, it's better to have a `provider` props to know the enabled provider. This lookup way will have confilct as the order.

Maybe a optional props with provider,if have this value to just get the model card, or use this lookup way.

File: src/config/featureFlags/schema.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ export const FeatureFlagsSchema = z.object({
   dalle: z.boolean().optional(),
   speech_to_text: z.boolean().optional(),
   token_counter: z.boolean().optional(),
-
+  model_name: z.boolean().optional(),
```
Comment (arvinxx) [inline src/config/featureFlags/schema.ts]: `model_name` is not clear enough for this feature. 

I perfer to `model_tag_display_mode` and with an enum. 

The default value should be `model_id`, and your perference is `model_name` . Maybe we can have another value for `provider/model_id` to show `openai/gpt-4o` in the furture

File: src/config/featureFlags/schema.ts [hunk 4]
```diff
@@ -100,6 +101,7 @@ export const mapFeatureFlagsEnvToState = (config: IFeatureFlags) => {
 
     showMarket: config.market,
     enableSTT: config.speech_to_text,
+    useModelName: config.model_name,
```
Comment (arvinxx) [inline src/config/featureFlags/schema.ts]: key can be `modelTagUseModelName`, value is `config.model_tag_display_mode === 'model_name'`

**Review by arvinxx — CHANGES_REQUESTED** [review]

File: src/config/featureFlags/schema.ts (hunk 3 above)
Comment (phuctm97) [reply src/config/featureFlags/schema.ts]: But feature flags only support boolean values now, doesn't it?

Comment (phuctm97) [reply src/config/featureFlags/schema.ts]: This docs doesn't mention how to set a enum value for a feature flag: https://lobehub.com/docs/self-hosting/advanced/feature-flags

Comment (arvinxx) [reply src/config/featureFlags/schema.ts]: > But feature flags only support boolean values now, doesn't it?

it seems yes. But we can extend it just like the sytnax of `OPENAI_MODEL_LIST` : https://lobehub.com/docs/self-hosting/advanced/model-list. 

`+key=value`

**Comment by khanhduyvt0101** [comment]
I just updated base on your review @arvinxx

**Comment by arvinxx** [comment]
@khanhduyvt0101 hi, please rebase to main branch and I will merge it.

**Comment by khanhduyvt0101** [comment]
I rebased @arvinxx

File: docs/self-hosting/advanced/auth/next-auth/casdoor.mdx [hunk 5]
```diff
@@ -50,7 +50,7 @@ There are also some optional fields that can enhance user experience:
 - Form CSS, Form CSS (Mobile):
 
   ```html
-  <style>
+  <Steps>
```
Comment (arvinxx) [inline docs/self-hosting/advanced/auth/next-auth/casdoor.mdx:53]: please keep the previous style tag

File: docs/self-hosting/advanced/auth/next-auth/casdoor.zh-CN.mdx [hunk 6]
```diff
@@ -48,7 +48,7 @@ tags:
 - 表单 CSS、表单 CSS（移动端）：
 
   ```html
-  <style>
+  <Steps>
```
Comment (arvinxx) [inline docs/self-hosting/advanced/auth/next-auth/casdoor.zh-CN.mdx:51]: here the same

File: docs/self-hosting/server-database.mdx [hunk 7]
```diff
@@ -140,5 +140,3 @@ For detailed configuration guidelines on S3, please refer to [S3 Object Storage]
 ## Getting Started with Deployment
 
 The above is a detailed explanation of configuring LobeChat with a server-side database. You can configure it according to your actual situation and then choose a deployment platform that suits you to start deployment:
-
-<PlatformCards urlPrefix={'server-database'} />
```
Comment (arvinxx) [inline docs/self-hosting/server-database.mdx:144]: revert the change

**Review by arvinxx — COMMENTED** [review]
it's OK with left. please update as comments
